### PR TITLE
Track if events, venues, organizers, and attendees were generated

### PR DESCRIPTION
**Pull request by borkweb**
Reset (for events, venues, and organizers) will now only delete generated posts. If you specify `--all`, it'll clear both generated and non generated alike.

### CODE REVIEW

File: src/Events/Generator/API.php [hunk 1]
```diff
@@ -153,6 +153,11 @@ public function generate_event() {
 
 		$event_id = tribe_create_event( $args );
 
+		// mark event, venue, and organizers as generated from tribe-cli
+		add_post_meta( $event_id, '_tribe_cli_generated', 1 );
```
Comment (bordoni) [inline src/Events/Generator/API.php]: Set this as a String Variable on the class. `_tribe_cli_generated`

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Events/Generator/API.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/Events/Generator/API.php]: Does that Boolean value really need to be that complicated?

**Review by sc0ttkclark — APPROVED** [review]

File: src/Events/Generator/API.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/Events/Generator/API.php]: Oh maybe you're talking about the meta key string itself :)

**Comment by borkweb** [comment]
Rather than setting the same value in multiple classes - which seemed just as _meh_ as hardcoding, I decided to set it in the container. That _did_ require globalizing the container, but that felt semi-ok. If anyone has a better idea, I'm open to it.

**Comment by lucatume** [comment]
Once we walk the globalizing road we will walk it all the way; the globalization of the container can be avoided (and the consequences of it with it) using a class as an Information Repository; e.g.:

```php
class Meta_Keys {
	public static $meta_key_one = 'one';
	public static $meta_key_two = 'two';
	public static $meta_key_three = 'three';
}
```

The problem we are trying to solve is having one place only to access an information; I'd like to avoid the Service Locator pattern (globalizing the DI container) as we are merely "hiding" a dependency and walk instead of the road of providing a class that only does a thing: provide information.

**Comment by borkweb** [comment]
Cool! Implemented.

**Review by lucatume — APPROVED** [review]

**Pull request merged**